Source/WebCore/Modules/webxr/WebXROpaqueFramebuffer.cpp

```diff
@@ -137,7 +137,7 @@ void WebXROpaqueFramebuffer::startFrame(const PlatformXR::Device::FrameData::Lay
     // Tell the GraphicsContextGL to use the IOSurface as the backing store for m_opaqueTexture.
     if (data.isShared) {
 #if !PLATFORM(IOS_FAMILY_SIMULATOR)
-        auto surfaceTextureAttachment = gCGL->attachIOSurfaceToSharedTexture(textureTarget, data.surface.get());
+        auto surfaceTextureAttachment = gCGL->attachIOSurfaceToExternalImage(textureTarget, data.surface.get());
```

Comment:
I believe if you're now abstracting the call, you want to invoke it through gl and not through gCGL.

So you would have:
```
gl.attachToExternalImage(textureTarget,  *data.surface);
```

The in-process variant of `GraphicsContextGL::attachToExternalImage(..., WebCore::IOSurface&)` needs to be implemented twice: 
- once for current code-path in GraphicsContextGLCocoa
- once for RemoteGraphicsContextGLProxyCocoa, obtaining the mach send right from the iosurface.

Reply:
But we can't create an in process IOSurface when RemoteGraphicsContextGLProxyCocoa is in use, can we?

Reply:
> But we can't create an in process IOSurface when RemoteGraphicsContextGLProxyCocoa is in use, can we?

Yeah, we can. So "WebGL GPUP Yes/No" is distinct choice to "IOKit blocking".

And note: the question is not logical, there is need in *creating* in-process IOSurface for RemoteGraphicsContextGLProxyCocoa.

The client gives use either send right to IOSurface or the IOSurface instance.

The cases with RemoteGraphicsContextGLProxyCocoa are:
 - If we already have in process IOSurface, we can give that to RemoteGraphicsContextGLProxyCocoa via obtaining send right to it.
 - If we already have Mach send right to a IOSurface, we can give that to RemoteGraphicsContextGLProxyCocoa.